src/3rd_party-static/jsoncpp/src/lib_json/json_reader.cpp

```diff
@@ -164,11 +166,12 @@ Reader::parse( const char *beginDoc, const char *endDoc,
 
 
 bool
-Reader::readValue()
+Reader::readValue(int *const depth_level )
 {
    Token token;
    skipCommentTokens( token );
    bool successful = true;
+   const int  max_allowed_depth_level = 50;
```

Comment:
@ZhdanovP why 50? Is it known at which depth level seg fault occurs? I don't think that 50 is a maximum

Reply:
@AKalinich-Luxoft actually we don't have any requirements for the depth as I know. We use 50 in Sync3 this cause that I took 50